src/main/java/com/thebois/models/world/structures/AbstractStructure.java

```diff
@@ -14,114 +15,84 @@
 
     private Position position;
     private StructureType structureType;
-    private Collection<IItem> allNeededItems = new ArrayList<>();
-    private Collection<IItem> deliveredItems = new ArrayList<>();
+    private IInventory allNeededItems;
```

Comment:
You could have this as just a list of item types required, so you don't need to compare the difference between two inventories.